File: evidence/capture-screenshots.js
const { chromium } = require('playwright');
const fs = require('fs');
const path = require('path');

const VIEWPORTS = [
  { name: 'iphone-se', width: 375, height: 667 },
  { name: 'iphone-14-pro', width: 390, height: 844 },
  { name: 'android', width: 360, height: 800 }
];

const ARTWORKS = [
  { id: '1', name: '星月夜' },
  { id: '2', name: '蒙娜丽莎' },
  { id: '3', name: '戴珍珠耳环的少女' },
  { id: '4', name: '神奈川冲浪里' },
  { id: '5', name: '最后的晚餐' }
];

const OUTPUT_DIR = path.join(__dirname, 'ui-artwork-info');
const BASE_URL = 'http://127.0.0.1:8889/index.html';

async function captureScreenshots() {
  const browser = await chromium.launch({ headless: true });
  const results = [];

  for (const artwork of ARTWORKS) {
    for (const viewport of VIEWPORTS) {
      const context = await browser.newContext({
        viewport: { width: viewport.width, height: viewport.height }
      });
      const page = await context.newPage();

      const url = `${BASE_URL}#work-${artwork.id}`;
      console.log(`Capturing: ${artwork.name} @ ${viewport.name} (${viewport.width}x${viewport.height})`);

      try {
        await page.goto(url, { waitUntil: 'networkidle', timeout: 30000 });
        await page.waitForTimeout(2000);

        const filename = `artwork-${artwork.id}-${viewport.name}.png`;
        const filepath = path.join(OUTPUT_DIR, filename);
        await page.screenshot({ path: filepath, fullPage: false });

        results.push({
          artworkId: artwork.id,
          artworkName: artwork.name,
          viewport: viewport.name,
          viewportSize: `${viewport.width}x${viewport.height}`,
          url: url,
          screenshot: filename,
          timestamp: new Date().toISOString()
        });

        console.log(`  -> ${filename}`);
      } catch (err) {
        console.error(`  FAILED: ${err.message}`);
        results.push({
          artworkId: artwork.id,
          artworkName: artwork.name,
          viewport: viewport.name,
          error: err.message
        });
      }

      await context.close();
    }
  }

  await browser.close();

  const manifestPath = path.join(OUTPUT_DIR, 'artwork-info-card-screenshots.json');
  fs.writeFileSync(manifestPath, JSON.stringify({
    generatedAt: new Date().toISOString(),
    totalScreenshots: results.filter(r => !r.error).length,
    totalExpected: ARTWORKS.length * VIEWPORTS.length,
    results: results
  }, null, 2));

  console.log(`\nManifest written to: ${manifestPath}`);
  return results;
}

captureScreenshots().catch(console.error);

File: evidence/capture-screenshots.py
#!/usr/bin/env python3
"""
Capture DevTools-style screenshots for typography evidence.
Run this script while the local server is running on port 8765.
Requires: playwright (pip install playwright)
"""

import subprocess
import sys
import os

WORKSPACE = "/Users/david/人文/艺术手册/artbook/.worktrees/t_be1a5c90"
EVIDENCE_DIR = os.path.join(WORKSPACE, "evidence")
SERVER_URL = "http://127.0.0.1:8765"

# Pages to capture
PAGES = [
    ("feed", "/", "Feed 页 - 品牌标题与作品列表"),
    ("detail", "/#/work/monet-1840-1", "Detail 页 - 作品标题与正文"),
    ("collection", "/#/collection/impressionism", "Collection 页 - 版心一致性"),
]

def check_playwright():
    """Check if playwright is installed"""
    try:
        import playwright
        return True
    except ImportError:
        return False

def capture_with_playwright():
    """Use Playwright to capture screenshots with viewport 390x844"""
    from playwright.sync_api import sync_playwright
    
    os.makedirs(EVIDENCE_DIR, exist_ok=True)
    
    with sync_playwright() as p:
        browser = p.webkit.launch(headless=True)
        page = browser.new_page(
            viewport={"width": 390, "height": 844},
            device_scale_factor=2
        )
        
        for name, path, desc in PAGES:
            url = SERVER_URL + path
            print(f"Capturing {name}: {url}")
            page.goto(url, wait_until="networkidle")
            screenshot_path = os.path.join(EVIDENCE_DIR, f"typography-{name}-390x844.png")
            page.screenshot(path=screenshot_path, full_page=True)
            print(f"  Saved: {screenshot_path}")
        
        browser.close()
    
    print("\n✓ All screenshots captured successfully")
    print(f"Output directory: {EVIDENCE_DIR}")

def main():
    if not check_playwright():
        print("Playwright not installed. Installing...")
        subprocess.check_call([sys.executable, "-m", "pip", "install", "playwright", "-q"])
        subprocess.check_call([sys.executable, "-m", "playwright", "install", "webkit"])
    
    capture_with_playwright()

if __name__ == "__main__":
    main()
